test/dist_autograd_test.py

```diff
@@ -126,13 +222,25 @@ def test_autograd_functions(self):
             )
             self.assertEqual(next_funcs[0][0], next_funcs[1][0])
 
-        # autograd context should be cleaned up by now.
-        with self.assertRaises(RuntimeError):
-            ctx = dist_autograd._retrieve_context(context_id)
+            # For send function when returning resonpose to client
+            # next function of the send function is the recv function
+            # for received tensor result returned from nested call
+            next_funcs = list(send_functions.values())[1].next_functions
+            self.assertEqual(1, len(next_funcs))
+            self.assertEqual(
+                "torch::distributed::autograd::RecvRpcBackward", next_funcs[0][0].name()
```

Comment:
(This does not need to be done in this PR)

It seems autograd function checks are scattered in several places, making it a little difficult to track. Is it possible to implement one general autograd function checking utility method which takes a `ctx` and a nested list/tuple/dict (or we could just add our own graph structure) `expected_graph`. Then a test could do the forward pass, then construct `expected_graph` right next the forward pass (so that it will be much easier to verify they match), then call the generic autograd function checking method to verify correctness.